tests/orm/test_querybuilder.py

```diff
@@ -1199,6 +1199,24 @@ def test_joins_group_node(self):
         for curr_id in [n1.pk, n2.pk, n3.pk, n4.pk]:
             assert curr_id in id_res
 
+    def test_joins_group_node_distinct(self):
+        """Test that when protecting only the group for a join on nodes, only unique groups are returned.
+
+        Regression test for #5535
+        """
+        group = orm.Group(label='mygroup').store()
+        node_a = orm.Data().store()
+        node_b = orm.Data().store()
+        group.add_nodes([node_a, node_b])
+
```

Comment:
I think it may be useful to keep the reverse order as well, even though that was not affected by this particular bug, it might be in the future:
```suggestion

        # First join the group on the data
        query = orm.QueryBuilder()
        query.append(orm.Group, project='id', tag='group')
        query.append(orm.Data, with_group='group')
        query.distinct()

        assert query.count() == 1
        assert query.all(flat=True) == [group.pk]
        
        # Then reverse and join the data on the group
```